syntax = "proto3";

service EdgeServer {
  // 用户通过此接口查询数据
  rpc ExecuteSQL(SQLRequest) returns (SQLResponse);

  // 边缘服务器定期上报请求统计
  rpc ReportStatistics(StatisticsRequest) returns (StatisticsResponse);

  // 中心服务器向边缘服务器下发缓存任务
  rpc UpdateCache(CacheUpdateRequest) returns (CacheUpdateResponse);
}

message SQLRequest {
  string query = 1;
  string data_source_id = 2; // 数据源Id
  int64 timestamp = 3;       // 查询时间戳
}

message SQLResponse {
  string result = 1;
}

message StatisticsRequest {
  repeated AccessRecord access_records = 1; // 访问记录列表
}

message AccessRecord {
  string data_source_id = 1;
  int64 timestamp = 2;
}

message StatisticsResponse {
  string status = 1;
}

message CacheUpdateRequest {
  repeated string queries = 1;
}

message CacheUpdateResponse {
  string status = 1;
}
